File: Sample/Scripts/Global/Gameplay/GameplayEventData.lua
local GameplayEventData = {}

GameplayEventData.instigator = nil
GameplayEventData.target = nil
GameplayEventData.eventTag = ""
GameplayEventData.payload = {}

function GameplayEventData:init(values)
    values = values or {}
    self.instigator = values.instigator
    self.target = values.target
    self.eventTag = values.eventTag or ""
    self.payload = values.payload or {}
end

return class(GameplayEventData)

File: Game/Engine/Source/Global/src/Gameplay/GameplayEventData.cpp
#include <Gameplay/GameplayEventData.hpp>
#include "GameplayValueUtils.hpp"

#include <cmath>
#include <stdexcept>
#include <utility>

GameplayEventData::GameplayEventData(RuntimeValue eventInstigator,
                                     RuntimeValue eventTarget, std::string tag,
                                     RuntimeIdentityPtr eventPayload)
    : instigator(std::move(eventInstigator)),
      target(std::move(eventTarget)),
      eventTag(std::move(tag)),
      payload(std::move(eventPayload)) {
    if (payload == nullptr) {
        payload = ludork::global::gameplay_detail::runtimeMap();
    }
}
